weave/Scripts/InputSources/Lobby.cs

```diff
@@ -27,21 +47,36 @@ public void Join(IInputSource inputSource)
 
         _playerInfos.Add(playerInfo);
         UpdatePlayerInfos();
+        PlayerJoinedListeners?.Invoke(inputSource);
     }
 
     public void Leave(IInputSource inputSource)
     {
         _playerInfos.RemoveWhere(info => info.InputSource.Equals(inputSource));
         UpdatePlayerInfos();
+        PlayerLeftListeners?.Invoke(inputSource);
     }
 
+    private static string GenerateLobbyCode(string allowedCharacters, int length)
+    {
+        var rnd = new Random();
+        var result = new char[length];
+        for (var i = 0; i < length; i++)
+            result[i] = allowedCharacters[rnd.Next(allowedCharacters.Length - 1)];
```

Comment:
What if this code is taken 🤭 (probably not a concern for us, fix if you want to)